frontend/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -14,465 +14,300 @@
+          if (currentEditingState !== currentWidgetState) {
+            widgetMgr.setStringValue(
+              element as WidgetInfo,
+              currentEditingState,
+              {
+                fromUi: true,
+              }
+            )
+          }
+        })()
+      }
     },
-    [sort, columns]
+    [widgetMgr, element, numRows]
   )
 
-  // Automatic table height calculation: numRows +1 because of header, and +2 pixels for borders
-  let maxHeight = Math.max(
-    (numRows + 1) * ROW_HEIGHT + 1 + 2,
-    MIN_TABLE_HEIGHT
+  const { onCellEdited, onPaste, onRowAppended, onDelete } = useDataEditor(
+    columns,
+    element.editingMode !== ArrowProto.EditingMode.DYNAMIC,
+    editingState,
+    getCellContent,
+    getOriginalIndex,
+    refreshCells,
+    applyEdits
   )
-  let initialHeight = Math.min(maxHeight, DEFAULT_TABLE_HEIGHT)
 
-  if (element.height) {
-    // User has explicitly configured a height
-    initialHeight = Math.max(element.height, MIN_TABLE_HEIGHT)
-    maxHeight = Math.max(element.height, maxHeight)
-  }
+  const { columns: glideColumns, onColumnResize } = useColumnSizer(
+    columns.map(column => toGlideColumn(column))
+  )
 
-  if (containerHeight) {
-    // If container height is set (e.g. when used in fullscreen)
-    // The maxHeight and height should not be larger than container height
-    initialHeight = Math.min(initialHeight, containerHeight)
-    maxHeight = Math.min(maxHeight, containerHeight)
+  const {
+    rowHeight,
+    minHeight,
+    maxHeight,
+    minWidth,
+    maxWidth,
+    resizableSize,
+    setResizableSize,
+  } = useTableSizer(
+    element,
+    numRows,
+    containerWidth,
+    containerHeight,
+    isFullScreen
+  )
 
-    if (!element.height) {
-      // If no explicit height is set, set height to max height (fullscreen mode)
-      initialHeight = maxHeight
-    }
-  }
+  // This is used as fallback in case the table is empty to
+  // insert cells indicating this state:
+  const getEmptyStateContent = React.useCallback(
+    ([_col, _row]: readonly [number, number]): GridCell => {
+      return {
+        ...getTextCell(true, false),
+        displayData: "empty",
+        contentAlign: "center",
+        allowOverlay: false,
+        themeOverride: {
+          textDark: theme.textLight,
+        },
+        span: [0, Math.max(columns.length - 1, 0)],
+      } as GridCell
+    },
+    [columns]
+  )
 
-  let initialWidth: number | undefined // If container width is undefined, auto set based on column widths
-  let maxWidth = containerWidth
-
-  if (element.useContainerWidth) {
-    // Always use the full container width
-    initialWidth = containerWidth
-  } else if (element.width) {
-    // User has explicitly configured a width
-    initialWidth = Math.min(
-      Math.max(element.width, MIN_TABLE_WIDTH),
-      containerWidth
+  // This is required for the form clearing functionality works:
```

Comment:
maybe instead of
```// This is required for the form clearing functionality works:```
```// This is required for the form clearing functionality:```